### DISCUSSION TITLE
Missing .dll file while calling .mexw64 in Simulink

### DISCUSSION THREAD

**Discussion by maskros-m** [casadi/casadi#3879] [Q&A]
Hi developers and users

I have been trying to include a C code in a simulink model via s-function, the .mex file was built up from the C generated code by Casadi (the code includes also fatrop library) and the wrapper as per https://web.casadi.org/blog/s-function/

My problem is, while running the simulink model, it always results in error: "The .mex file is not found". It has been said that some libraries are missing, while scanning the dependencies of .mex file. Unfortunately there are many missing .dll that cannot be found anywhere. Something like in the picture. May I ask whether someone has encountered similar issue and please give me some hints?
 
![Unbenannt](https://github.com/user-attachments/assets/176b3a76-4b76-4b16-8d0f-a809a326114d)

Thanks in advance!
Best regards,
Anh

**Comment by jgillis** [comment]
The ones in the screenshot are mostly false positives. libmw is the standard prefix for mathworks libraries. In the Simulink context they will surely be found. Try copying fatrop.dll and blasfeo.dll to a location where they can be found by the 'loader'. Perhaps in the same directory as you mex file. Or, restart Matlab with the PATH env variable set to their location.

**Reply by maskros-m** [reply]
Thanks for your quick reply! Indeed, I have the same impression that s a false alarm. However, Simulink always complains the .mex file is not found, also when a copy of both libfatrop.dll and libblasfeo.dll is in the same folder with .mex file.

![Unbenannt](https://github.com/user-attachments/assets/f2a63b8c-c911-436b-a597-0dfb7133cf59)

**Reply by jgillis** [reply]
So perhaps try adding this directory to the PATH (windows wide) and restart matlab

**Reply by maskros-m** [reply]
Hi, I have added the path of this directory to System Variable Environment, but the error is still persistent.

**Reply by maskros-m** [reply]
Adding the path of casadi in System Variable Environment resolves my issue. Thanks for your support!
